datafusion/core/src/datasource/listing_table_factory.rs

```diff
@@ -67,12 +67,23 @@ impl TableProviderFactory for ListingTableFactory {
         let file_extension = get_extension(cmd.location.as_str());
 
         let file_format: Arc<dyn FileFormat> = match file_type {
-            FileType::CSV => Arc::new(
-                CsvFormat::default()
-                    .with_has_header(cmd.has_header)
-                    .with_delimiter(cmd.delimiter as u8)
-                    .with_file_compression_type(file_compression_type),
-            ),
+            FileType::CSV => {
+                let mut statement_options = StatementOptions::from(&cmd.options);
+                let quote = statement_options
+                    .take_str_option("quote")
+                    .map_or(b'"', |x| x.as_bytes()[0]);
```

Comment:
This effectively overrides the default values in the CSV format.

What would you think about only setting the values if it was different than the default

Something like
```rust
   let mut format = CsvFormat::default();
   if let Some(quote) = statement_options.take_str_option("quote") {
    format = format.with_quote(quote)
   }
   ...
```

?

Reply:
you are right